collect_app/src/main/java/org/odk/collect/android/widgets/BaseStringWidget.java

```diff
@@ -154,4 +158,37 @@ protected Double getDoubleAnswerValue() {
         }
         return d;
     }
+
+    protected void setDisplayDecimalValueFromModel() {
```

Comment:
Ok, now I'm getting really uncomfortable with how much type-specific behavior ends up in the base class! Could you use polymorphism here? That is, keep `setDisplayValueFromModel` abstract here and introduce a common base class for each of the string, integer and decimal class pairs to share the `setDisplayValueFromModel` implementation between the two?